runtime/binding-asyncapi/src/main/java/io/aklivity/zilla/runtime/binding/asyncapi/internal/config/AsyncapiProtocol.java

```diff
@@ -113,54 +111,41 @@ protected <C> CatalogedConfigBuilder<C> injectJsonSchemas(
+            !asyncapi.components.messages.isEmpty())
         {
-            AsyncapiMessage firstAsyncapiMessage = asyncApi.components.messages.entrySet().stream()
+            AsyncapiMessage firstAsyncapiMessage = asyncapi.components.messages.entrySet().stream()
                 .findFirst().get().getValue();
-            contentType = AsyncapiMessageView.of(asyncApi.components.messages, firstAsyncapiMessage).contentType();
+            contentType = AsyncapiMessageView.of(asyncapi.components.messages, firstAsyncapiMessage).contentType();
         }
-        return contentType != null && jsonContentType.reset(contentType).matches() || asyncApi.defaultContentType != null &&
-            jsonContentType.reset(asyncApi.defaultContentType).matches();
+        return contentType != null && jsonContentType.reset(contentType).matches() || asyncapi.defaultContentType != null &&
+            jsonContentType.reset(asyncapi.defaultContentType).matches();
     }
 
     protected abstract boolean isSecure();
 
     protected Map<String, String> resolveSecuritySchemes()
     {
-        requireNonNull(asyncApi);
+        requireNonNull(asyncapis);
         Map<String, String> result = new HashMap<>();
-        if (asyncApi.components != null && asyncApi.components.securitySchemes != null)
+        for (Asyncapi asyncapi : asyncapis)
         {
-            for (String securitySchemeName : asyncApi.components.securitySchemes.keySet())
+            if (asyncapi.components != null && asyncapi.components.securitySchemes != null)
             {
-                String guardType = asyncApi.components.securitySchemes.get(securitySchemeName).bearerFormat;
-                //TODO: change when jwt support added for mqtt in asyncapi
-                //if ("jwt".equals(guardType))
-                //{
-                //    result.put(securitySchemeName, guardType);
-                //}
-                result.put(securitySchemeName, guardType);
-            }
-        }
-        return result;
-    }
-
-    protected URI findFirstServerUrlWithScheme(
-        String scheme)
-    {
-        requireNonNull(scheme);
-        URI result = null;
-        for (String key : asyncApi.servers.keySet())
-        {
-            AsyncapiServerView server = AsyncapiServerView.of(asyncApi.servers.get(key));
-            if (scheme.equals(server.url().getScheme()))
-            {
-                result = server.url();
-                break;
+                for (String securitySchemeName : asyncapi.components.securitySchemes.keySet())
+                {
+                    String guardType = asyncapi.components.securitySchemes.get(securitySchemeName).bearerFormat;
+                    //TODO: change when jwt support added for mqtt in asyncapi
+                    //if ("jwt".equals(guardType))
+                    //{
+                    //    result.put(securitySchemeName, guardType);
+                    //}
```

Comment:
Is this TODO still relevant?

Reply:
I think yes. As asyncapi doesn't have JWT support added for MQTT yet and we use this to determine if we need to add the mqtt authorization options for the server. If we filter for "jwt" we'll never add those options.